pkgs/servers/mautrix-facebook/default.nix

```diff
@@ -0,0 +1,59 @@
+{
+  enableSystemd ? stdenv.isLinux,
+  fetchFromGitHub,
+  lib,
+  python3,
+  stdenv,
+}:
+
+python3.pkgs.buildPythonPackage rec {
+  pname = "mautrix-facebook";
+  version = "v0.3.1";
+
+  src = fetchFromGitHub {
+    owner = "tulir";
+    repo = pname;
+    rev = version;
+    sha256 = "0m7nznx3z6cg4wgvjybdivx22ifxcdri4i8501yibsri0jnpf0y2";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    CommonMark
+    aiohttp
+    asyncpg
+    mautrix
+    paho-mqtt
+    pillow
+    prometheus-client
+    pycryptodome
+    python-olm
+    python_magic
+    ruamel_yaml
+    unpaddedbase64
+    yarl
+  ] ++ lib.optional enableSystemd systemd;
+
+  doCheck = false;
+
+  prePatch = ''
+    sed -ie 's/^asyncpg.*/asyncpg>=0.20/' requirements.txt
```

Comment:
Is it possible to be a little more precise here? or make a PR for upstream?

Reply:
The ^ makes it fairly precise. Upstream for some reason sets max versions but I can ask them to reconsider that. They really recommend virtuanlenv so I don't know if they will be receptive.

Reply:
Sure sure, I don't know much about python packaging so whatever you think is best.

Reply:
Please use substituteInPlace and postPatch.

Reply:
substituteInPlace wont' do regexes so I wasn't able to use it.